test/test_sparse.py

```diff
@@ -2881,6 +2882,136 @@ def sparse_log(x):
         test_op(3, 100, [3, 4, 2, 3, 5, 2])
         test_op(4, 100, [3, 4, 2, 3, 5, 2])
 
+    @skipIfRocm
+    def test_sparse_matmul(self):
+        """
+        This function test `torch.sparse.mm` when both the mat1 and mat2 are sparse tensors. 
+        """
+
+        def _indices2csr(indices, dim):
+            nnz = len(indices)
+            r = [0] * (dim + 1)
+            last_i = 0
+            for i in indices:
+                if i != last_i:
+                    for _i in range(last_i, i + 1):
+                        r[_i + 1] = r[last_i + 1]
+                    last_i = i
+                r[last_i + 1] += 1
+            for _i in range(last_i, dim):
+                r[_i + 1] = r[last_i + 1]
+            assert r[-1] == nnz
+            return r
+
+        def sparse_mm(a, b):
```

Comment:
I would suggest introducing, say, `method` keyword argument to `sparse_mm` such that `method in ['scipy', 'direct']`, and then eliminate the `try-except` construction below. This would allow testing if `direct` gives the same result as `scipy` (when available).

Reply:
With try-except, it will be more difficult to track down any bugs because it is not clear from reading the code which path is used (try block or except block).